examples/vision/melanoma_classification_with_tfrecords.py

```diff
@@ -0,0 +1,438 @@
+
+We also reshape our data so that all of the images will be the same shape. Although the
+TFRecord files have already been reshaped for us, it is best practice to reshape the
+input so that we know exactly what's going in to our model.
+"""
+
+
+def decode_image(image):
+    image = tf.image.decode_jpeg(image, channels=3)
+    image = tf.cast(image, tf.float32) / 255.0
+    image = tf.reshape(image, [*IMAGE_SIZE, 3])
+    return image
+
+
+"""
+As we load in our data, we need both our ```X``` and our ```Y```. The X is our image; the
+model will find features and patterns in our image dataset. We want to predict Y, the
+probability that the lesion in the image is malignant. When we input our training dataset
+into the model, it is necessary know what the labels of our images are. For our testing
+set, we do not need to feed in the labels because the model will predict it for us. In
+this particular dataset, our label is the ```target``` feature. We will to through our
+TFRecords and parse out the image and the target values.
+"""
+
+
+def read_tfrecord(example, labeled):
+    tfrecord_format = (
+        {
+            "image": tf.io.FixedLenFeature([], tf.string),
+            "target": tf.io.FixedLenFeature([], tf.int64),
+        }
+        if labeled
+        else {
+            "image": tf.io.FixedLenFeature([], tf.string),
+            "image_name": tf.io.FixedLenFeature([], tf.string),
+        }
+    )
+    example = tf.io.parse_single_example(example, tfrecord_format)
+    image = decode_image(example["image"])
+    if labeled:
+        label = tf.cast(example["target"], tf.int32)
+        return image, label
+    idnum = example["image_name"]
+    return image, idnum
+
+
+"""
+Our dataset is not ordered in any meaningful way, so the order can be ignored when
+loading our dataset. By ignoring the order and reading files as soon as they come in, it
+will take a shorter time to load the data.
+"""
+
+
+def load_dataset(filenames, labeled=True, ordered=False):
+    ignore_order = tf.data.Options()
+    if not ordered:
+        ignore_order.experimental_deterministic = False  # disable order, increase speed
+    dataset = tf.data.TFRecordDataset(
+        filenames, num_parallel_reads=AUTOTUNE
+    )  # automatically interleaves reads from multiple files
+    dataset = dataset.with_options(
+        ignore_order
+    )  # uses data as soon as it streams in, rather than in its original order
+    dataset = dataset.map(
+        partial(read_tfrecord, labeled=labeled), num_parallel_calls=AUTOTUNE
+    )
+    # returns a dataset of (image, label) pairs if labeled=True or (image, id) pairs if labeled=False
+    return dataset
+
+
+"""
+## 2.2 Data augmentation
+
+Image augmentation introduces new images into our data, and generally prevents the model
+from overfitting and leads to higher accuracy on testing datasets. For this particular
+dataset, we will randomly flip the image horizontally.
+"""
+
+
+def augmentation_pipeline(image, label):
```

Comment:
You would typically apply augmentation via preprocessing layers (applied after batching). See: https://keras.io/guides/preprocessing_layers/#image-data-augmentation-ondevice

Or https://keras.io/examples/vision/image_classification_from_scratch/#using-image-data-augmentation